runtime/oti/VMHelpers.hpp

```diff
@@ -2232,6 +2232,82 @@ class VM_VMHelpers
 		targetThread->privateFlags2 |= J9_PRIVATE_FLAGS2_REENTER_INTERPRETER;
 		indicateAsyncMessagePending(targetThread);
 	}
+
```

Comment:
I don't think this code belongs here - it should be in a .cpp file (put it in the async message handler if you like) and called via the function table for the JCL caller.

Reply:
This comment is resolved, but it seems valid to me. Shall I address it?

Reply:
Updated.